### PR TITLE
Fix/improves button component

### PR DESCRIPTION
**Pull request by clayton-duarte**
- Improves button component and fix wrong styles;
- Fixes InputAutocomplete controls;
- Updates some component documentations;

### CODE REVIEW

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -1,64 +1,79 @@
...
-  /** Changes the style to have a border instead of a container-ish appearance */
-  outline?: boolean
-  /** Makes the button looks like a link */
-  link?: boolean
-  /** Makes the button have a slightly bigger height */
   big?: boolean
-  /** Makes the button background lighter and the text darker */
-  secondary?: boolean
-  /** Prevents form submit event when button is nested in a form */
-  noSubmit?: boolean
```
Comment (l0rdcafe) [inline src/components/Button/index.tsx:22]: removing this will probably break our checkout form

File: src/components/InputAutocomplete/index.mdx [hunk 2]
```diff
@@ -32,7 +34,7 @@ Hey! I'm a component documentation! Look how I'm easy to write and read.
   ]}
 />
 
-### INSET variant
+##### Inset - variant
```
Comment (l0rdcafe) [inline src/components/InputAutocomplete/index.mdx:37]: nit: can we keep it uppercase for consistency with the variant enum?

File: src/components/Button/index.tsx (hunk 1 above)
Comment (clayton-duarte) [reply src/components/Button/index.tsx:22]: yup, I'll fix that while upgrading OOF (basically adding `type="button"` at the component or `e.preventDefault()` at the handler)

File: src/components/InputAutocomplete/index.mdx (hunk 2 above)
Comment (clayton-duarte) [reply src/components/InputAutocomplete/index.mdx:37]: actually, we can, hehehe, updating in a sec

File: src/components/Button/index.tsx (hunk 1 above)
Comment (l0rdcafe) [reply src/components/Button/index.tsx:22]: cool just making sure we don't break things 😄

**Review by l0rdcafe — APPROVED** [review]

**Pull request merged**